Test failing for 'internal/task' package

**Is this a BUG REPORT or FEATURE REQUEST?**: Bug report
**NGINX Ingress controller version**: `master`
**Kubernetes version**: none
**Environment**: Linux 4.9.65-3+deb9u2, Debian stretch (container), Go 1.10

**What happened**:

I noticed the following test failed consistently in my CI environment (Jenkins w/ container step) when invoking `go test -race -cover $(PKG)/internal/...`:
```
E0313 18:17:50.185305    9583 queue.go:65] queue has been shutdown, failed to enqueue: {testKey testValue}
E0313 18:17:50.200777    9583 queue.go:73] failed to get key
--- FAIL: TestSkipEnqueue (0.01s)
	queue_test.go:154: sr should be 1, but is 2
FAIL
coverage: 84.8% of statements
FAIL	k8s.io/ingress-nginx/internal/task	0.353s
```

I can't reproduce locally. There is definitely something wrong with the execution environment but I need more time to investigate.

**What you expected to happen**:

All tests pass.

**How to reproduce it**:

Jenkinsfile:

```groovy
def goPackage = 'k8s.io/ingress-nginx'

ws("${JENKINS_HOME}/jobs/${JOB_NAME}/builds/${BUILD_ID}/src/${goPackage}") {
	withEnv(["GOPATH=${JENKINS_HOME}/jobs/${JOB_NAME}/builds/${BUILD_ID}", 'GOCACHE=/tmp/.gocache']) {

	docker.image('golang:1.10').inside {
		stage('Build'){
			sh 'CGO_ENABLED=0 go build -ldflags "-s -w" k8s.io/ingress-nginx/cmd/nginx'
		}
		stage('Test'){
			sh 'go test -race -cover k8s.io/ingress-nginx/internal/...'
		}
	}
}}
```

Equivalent:

```
$ docker run --name tests -t -d -u 9007:9007 --rm \
    -e GOCACHE=/tmp/.gocache \
    -v $(pwd):/go/src/k8s.io/ingress-nginx golang:1.10 \
    cat
$ docker exec tests \
    sh -c 'CGO_ENABLED=0 go build -ldflags "-s -w" k8s.io/ingress-nginx/cmd/nginx'
$ docker exec tests \
    sh -c 'go test -race -cover k8s.io/ingress-nginx/internal/...'
$ docker stop tests
```

--- a/internal/task/queue_test.go
+++ b/internal/task/queue_test.go
@@ -137,8 +137,6 @@ func TestSkipEnqueue(t *testing.T) {
 	atomic.StoreUint32(&sr, 0)
 	q := NewCustomTaskQueue(mockSynFn, mockKeyFn)
 	stopCh := make(chan struct{})
-	// run queue
-	go q.Run(time.Second, stopCh)
 	// mock object whichi will be enqueue
 	mo := mockEnqueueObj{
 		k: "testKey",
@@ -148,6 +146,8 @@ func TestSkipEnqueue(t *testing.T) {
 	q.Enqueue(mo)
 	q.Enqueue(mo)
 	q.Enqueue(mo)
+	// run queue
+	go q.Run(time.Second, stopCh)
 	// wait for 'mockSynFn'
 	time.Sleep(time.Millisecond * 10)
 	if atomic.LoadUint32(&sr) != 1 {

